
# Publish all joint states -----------------------------------
/motoman_gp8/joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50


# Trajectory Controllers
/motoman_gp8/gp8_controller:
#  type: position_controllers/JointTrajectoryController
  type: effort_controllers/JointTrajectoryController
  joints:
      - joint_1_s
      - joint_2_l
      - joint_3_u
      - joint_4_r
      - joint_5_b
      - joint_6_t
      - finger_joint

  gains:
    joint_1_s: { p: 400, d: 800, i: 5, i_clamp: 1 }
    joint_2_l: { p: 1200, d: 240, i: 5, i_clamp: 1 }  #p1200d240
    joint_3_u: { p: 800, d: 160, i: 5, i_clamp: 1 }  #p1200d240
    joint_4_r: { p: 300, d: 75, i: 5, i_clamp: 1 }
    joint_5_b: { p: 40, d: 12, i: 0.5, i_clamp: 0.1 }  # p80d12
    joint_6_t: { p: 8, d: 1.2, i: 0.5, i_clamp: 0.1 }
    finger_joint: { p: 40, d: 0.01, i: 0.0, i_clamp: 0.0}


#  gains:
#    joint_1_s: { p: 500, d: 100, i: 5, i_clamp: 1 }
#    joint_2_l: { p: 1500, d: 300, i: 5, i_clamp: 1 }  # p500d70
#    joint_3_u: { p: 1500, d: 300, i: 5, i_clamp: 1 }  # p500d70
#    joint_4_r: { p: 400, d: 85, i: 5, i_clamp: 1 }
#    joint_5_b: { p: 100, d: 15, i: 0.5, i_clamp: 0.1 }  # p40d2.5
#    joint_6_t: { p: 10, d: 1.5, i: 0.5, i_clamp: 0.1 }
#    finger_joint: { p: 50, d: 0.01, i: 0.0, i_clamp: 0.0}  #p50d0.05i0


